### PR TITLE
fix: multivariate polynomial rings over zero rings

### PR DESCRIPTION
**Pull request by thofma**

### CODE REVIEW

File: src/generic/GenericTypes.jl [hunk 1]
```diff
@@ -342,11 +342,12 @@ end
    ord::Symbol
    num_vars::Int
    N::Int
+   istrivial::Bool
 
    function MPolyRing{T}(R::Ring, s::Vector{Symbol}, ord::Symbol, N::Int,
                          cached::Bool = true) where T <: RingElement
       return get_cached!(MPolyID, (R, s, ord, N), cached) do
-         new{T}(R, s, ord, length(s), N)
+         new{T}(R, s, ord, length(s), N, is_zero(one(R)))
```
Comment (fingolfin) [inline src/generic/GenericTypes.jl:350]: Maybe this?
```suggestion
         new{T}(R, s, ord, length(s), N, is_trivial(R))
```

**Comment by thofma** [comment]
We are not getting the "expected" errors in Oscar, for which we need https://github.com/oscar-system/Oscar.jl/pull/4350

**Review by fingolfin — APPROVED** [review]
Overall this looks fine to me (ignoring whether or not this causes any upstream breakage -- but if we put this into 0.44.0, it should be OK?

The caching of `is_trivial`/`istrivial` may be unnecessary now, but it also doesn't hurt, so I don't mind!

**Comment by thofma** [comment]
Yes, I think it is fine and we also fixed all the issues in Oscar that this revealed. Oscar master branch is passing and the release branch is failing (as expected).

**Pull request merged**